mycroft/client/speech/hotword_factory.py

```diff
@@ -0,0 +1,143 @@
+#
+# This file is part of Mycroft Core.
+#
+# Mycroft Core is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Mycroft Core is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Mycroft Core.  If not, see <http://www.gnu.org/licenses/>.
+
+from mycroft.configuration import ConfigurationManager
+from mycroft.util.log import getLogger
+from os.path import dirname, exists, join, abspath
+import os
+import time
+import tempfile
+
+__author__ = 'seanfitz, jdorleans, jarbas'
+
+LOG = getLogger("HotwordFactory")
+
+BASEDIR = dirname(abspath(__file__)).join("recognizer")
+
+
+class HotWordEngine(object):
+    def __init__(self, key_phrase, config=None, lang="en-us"):
+        self.lang = str(lang).lower()
+        self.key_phrase = str(key_phrase).lower()
+        if config is None:
+            config = ConfigurationManager.get().get("hot_words", {})
+            config = config.get(self.key_phrase, {})
+        self.config = config
+        self.listener_config = ConfigurationManager.get().get("listener", {})
+
+    def found_wake_word(self, frame_data):
+        return False
+
+
+class PocketsphinxHotWord(HotWordEngine):
+    def __init__(self, key_phrase, config=None, lang="en-us"):
+        super(PocketsphinxHotWord, self).__init__(key_phrase, config, lang)
+        # Hotword module imports
+        from pocketsphinx import Decoder
+        # Hotword module config
+        module = self.config.get("module")
+        if module != "pocketsphinx":
+            LOG.warning(
+                str(module) + " module does not match with Hotword class "
+                         "pocketsphinx")
+        # Hotword module params
+        self.phonemes = self.config.get("phonemes", "HH EY . M AY K R AO F T")
+        self.threshold = self.config.get("threshold", 1e-90)
+        self.sample_rate = self.listener_config.get("sample_rate", 1600)
+        dict_name = self.create_dict(key_phrase, self.phonemes)
+        config = self.create_config(dict_name, Decoder.default_config())
+        self.decoder = Decoder(config)
+
+    def create_dict(self, key_phrase, phonemes):
+        (fd, file_name) = tempfile.mkstemp()
+        words = key_phrase.split()
+        phoneme_groups = phonemes.split('.')
+        with os.fdopen(fd, 'w') as f:
+            for word, phoneme in zip(words, phoneme_groups):
+                f.write(word + ' ' + phoneme + '\n')
+        return file_name
+
+    def create_config(self, dict_name, config):
+        model_file = join(BASEDIR, 'recognizer', 'model', self.lang, 'hmm')
```

Comment:
This is why the test is failing. Right now this gets turned into `join('.. speech/recognizer', 'recognizer', ...`.